### PR TITLE
[Data Discovery] Persist DD options into local and session storage

### PR DESCRIPTION
**Pull request by tfrcarvalho**
# Description

* Persist all DD options (e.g. UX config, filters search) into session storage
* Persist UX options (show/hide sidebars, current tab, selected columns) into local storage
* The new stored fields are not sensitive data

# Tests

* Load DD and change filters, search or UX options
* Click my data again or go to the report page and comeback, and see the same configuration
* Close the browser and reopen and see that only UX config changes are still active

### CODE REVIEW

File: app/assets/src/components/views/discovery/DiscoveryView.jsx [hunk 1]
```diff
@@ -159,20 +175,30 @@ class DiscoveryView extends React.Component {
...
     const urlState = pick(urlFields, this.state);
+    const localState = pick(localFields, this.state);
 
+    // Saving on URL enables sharing current view with other users
     let urlQuery = this.urlParser.stringify(urlState);
     if (urlQuery) {
       urlQuery = `?${urlQuery}`;
     }
 
+    // History state may include some fields that enable fast loading of previous pages (e.g. project)
```
Comment (jshoe) [inline app/assets/src/components/views/discovery/DiscoveryView.jsx]: this comment is a little confusing to me. "fast loading" as in direct loading of the previous pages?

Comment (tfrcarvalho) [reply app/assets/src/components/views/discovery/DiscoveryView.jsx]: Loading when clicking back or the history dropdown on the top-left corner of the browser.

Comment (tfrcarvalho) [reply app/assets/src/components/views/discovery/DiscoveryView.jsx]: Any suggestions for better wording?

**Comment by gregdingle** [comment]
Just a usability comment: in my experience it is usually a good idea to have a "reset to defaults" button when UI state is persistent.

**Comment by tfrcarvalho** [comment]
> Just a usability comment: in my experience it is usually a good idea to have a "reset to defaults" button when UI state is persistent.

I would agree that could be important for the field selection. But that would be a follow up change. 
Also, we are not sure our current defaults are the thing users want to get back to as default. So, I would first have a way to set the "default" per user.

File: app/assets/src/components/views/discovery/DiscoveryView.jsx (hunk 1 above)
Comment (jshoe) [reply app/assets/src/components/views/discovery/DiscoveryView.jsx]: resolved

**Pull request merged**